src/main/java/frc/robot/subsystems/LiftSubsystem.java

```diff
@@ -97,10 +116,10 @@ protected void initialize() {
             @Override
             protected void usePIDOutput(final double heightCorrection) {
                 double liftSpeed = heightCorrection;
-                if (upperLimit.get() && liftSpeed > 0) {
+                if (!upperLimit.get() && liftSpeed > 0) {
```

Comment:
Since you had to change all the `upperLimit.get()` to `!upperLimit.get()`, this would be a good place in the future to make a function `boolean isAtUpperLimit() { return !upperLimit.get(); }`. That way you only have to change it in once place.